linera-core/src/client/mod.rs

```diff
@@ -2080,6 +2078,7 @@ where
                 Ok(ExecuteBlockOutcome::Executed(certificate))
             }
             ClientOutcome::Committed(Some(certificate)) => {
+                // How could this have happened? We took the exclusive lock.
```

Comment:
Because it could be a multi-owner chain in multi-leader mode, and someone else proposed a block at the same time.